registry/service.go

```diff
@@ -131,16 +131,7 @@ func (s *DefaultService) Auth(ctx context.Context, authConfig *types.AuthConfig,
 	}
 
 	for _, endpoint := range endpoints {
-		status, token, err = loginV2(authConfig, endpoint, userAgent)
-		if err == nil {
-			return
-		}
-		if fErr, ok := err.(fallbackError); ok {
-			logrus.WithError(fErr.err).Infof("Error logging in to endpoint, trying next endpoint")
-			continue
-		}
-
-		return "", "", err
+		return loginV2(authConfig, endpoint, userAgent)
```

Comment:
I think this is where the problem is for the failing test; previously this was returning a `fallbackError` if;

- the registry didn't support `v2` (in which case it would fallback to `v1`
- connecting with the registry using `https` failed, and the registry we're connecting to is marked as `insecure registry`

In the latter case, it should continue trying the next (non-https) endpoint, and currently we only try the first endpoint in the list.

Not sure if we should continue trying on every type of error though (authentication failed errors should likely not be ignored)